backend-lucene/src/main/java/org/hibernate/search/v6poc/backend/lucene/types/codec/impl/LuceneFieldFieldCodec.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.v6poc.backend.lucene.types.codec.impl;
+
+import java.util.Objects;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+import org.apache.lucene.document.Document;
+import org.apache.lucene.index.IndexableField;
+import org.hibernate.search.v6poc.backend.lucene.document.impl.LuceneDocumentBuilder;
+
+public final class LuceneFieldFieldCodec<V> implements LuceneFieldCodec<V> {
+
+	private BiFunction<String, V, IndexableField> fieldProducer;
+
+	private Function<IndexableField, V> fieldValueExtractor;
+
+	public LuceneFieldFieldCodec(BiFunction<String, V, IndexableField> fieldProducer, Function<IndexableField, V> fieldValueExtractor) {
+		this.fieldProducer = fieldProducer;
+		this.fieldValueExtractor = fieldValueExtractor;
+	}
+
+	@Override
+	public void encode(LuceneDocumentBuilder documentBuilder, String absoluteFieldPath, V value) {
+		if ( value == null ) {
+			return;
+		}
+
+		documentBuilder.addField( fieldProducer.apply( absoluteFieldPath, value ) );
```

Comment:
Forgot to mention this: don't you think that, by default, we should run some sort of validation on the field name to prevent users to add `IndexableField` instances with the wrong path?
We could allow to disable those checks through a configuration property if you think it would be too expensive, but I really think checking that by default would avoid some unfortunate mistakes... Say, for example, the user could write `new StringField( "myField", value.toString() )` because they named their field `myField` and don't see why the `absoluteFieldPath` we give them would be different. And then when they use their bridge together with an `IndexedEmbedded`, KA-BOOM.